src/llmcompressor/transformers/finetune/session_mixin.py

```diff
@@ -79,15 +79,29 @@ def __init__(
 
         # parse training and metadata args
         training_args = kwargs.get("args")
-        self.metadata = (
-            self._extract_metadata(
+
+        self.metadata = None
+        if training_args is not None:
+            # trl_sft_trainer pathway. Both training_args and data_args
+            # have `max_seq_length` which causes collision error. This is the
+            # only shared parameter, where training arg is `TRLSFTConfig` that
+            # inherits HuggingFace's `TrainingArguments`
+            training_args_dict = training_args.to_dict()
+            if "max_seq_length" in training_args_dict:
+                training_args_dict["training_args_max_seq_length"] = (
+                    training_args_dict.pop("max_seq_length")
+                )
+                logger.warning(
```

Comment:
This warning confuses me, as the code makes no mention of the data args

Reply:
data_args is used in this class and also set in the trl example code!
This pathway doesnt go through parse args.

Reply:
Okay but the code doesn't actually override the value in data_args, so now there could be two values for seq length, one in data args and another in the metadata

Reply:
This is fine for now, but should really exist in SFTTrainer.__init__ since this processing is very specific to handling trl `args` in kwargs